src/sentry/sentry_metrics/kafka_metrics_interface.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Mapping, Optional, Sequence, Union
+
+from arroyo import Topic
+from arroyo.backends.abstract import Producer
+from arroyo.backends.kafka import KafkaPayload, KafkaProducer, build_kafka_configuration
+from django.conf import settings
+
+from sentry.sentry_metrics.metrics_interface import GenericMetricsBackend
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+from sentry.utils import json
+from sentry.utils.kafka_config import get_kafka_producer_cluster_options
+
+
+def build_mri(metric_name: str, type: str, use_case_id: UseCaseID, unit: Optional[str]) -> str:
+    mri_unit = "none" if unit is None else unit
+    return f"{type}:{use_case_id.value}/{metric_name}@{mri_unit}"
+
+
+# def set_future(f: Future[BrokerValue[KafkaPayload]]) -> Future[None]:
+#     new_future: Future[None] = Future()
+
+#     if f.exception() is not None:
+#         new_future.set_exception(f.exception())
+#     else:
+#         new_future.set_result(None)
+
+#     return new_future
+
+
+class KafkaMetricsBackend(GenericMetricsBackend):
+    def __init__(self) -> None:
+
+        kafka_topic_name = settings.KAFKA_INGEST_PERFORMANCE_METRICS
+        self.kafka_topic = Topic(kafka_topic_name)
+
+        kafka_topic_dict = settings.KAFKA_TOPICS[kafka_topic_name]
+        assert kafka_topic_dict is not None
+        cluster_name = kafka_topic_dict["cluster"]
+        producer_config = get_kafka_producer_cluster_options(cluster_name)
+        producer_config.pop("compression.type", None)
+        producer_config.pop("message.max.bytes", None)
+        self.producer: Producer = KafkaProducer(
+            build_kafka_configuration(default_config=producer_config)
```

Comment:
Mind double checking that `lz4` compression is correctly picked up from the settings here?

Reply:
yep, `"compression.type": "lz4"`

Reply:
I had to pop both of those keys though, because I get ValueErrors otherwise (`ValueError: The `message.max.bytes` configuration key is not supported.`)

Reply:
Please don't pop them off, we want those settings. Definitely the compression at least

Reply:
That's not the list that this code is hitting, [here](https://github.com/getsentry/arroyo/blob/main/arroyo/backends/kafka/configuration.py#L16-L59) is the list, which is in Arroyo and seems to not support compression. Should I change this list?

Reply:
This [code](https://github.com/getsentry/sentry/blob/master/src/sentry/issues/producer.py#L15-L20) in Sentry, which is also using the `KafkaProducer` from Arroyo, seems to pop off the same configs

Reply:
Yeah, add what you need to the list. (Or pass it through override_parama)

Reply:
I can do the override for now and maybe do the list update as a follow-up?